run-tests.php

```diff
@@ -3719,4 +3737,79 @@ function check_proc_open_function_exists()
     }
 }
 
+class OutputHandler
+{
+
+    /**
+     * @var null|string
+     */
+    private $htmlOutputFilename = null;
```

Comment:
```suggestion
    private ?string $htmlOutputFilename = null;
```
This one I'm less sure as I don't know how far back in compatibility we need to go, but if we are only targeting master this would make sense IMHO.

Reply:
Per the comment at the top of the file, this script is currently compatible with PHP 7.0. We could discuss raising the constraint, but unlikely to go to 7.4. (The test runner can be invoked with a different PHP version than it is testing. I believe @TysonAndre also expressed a desire to use newer test runners on PECL extensions that support old PHP versions, so it's possible to use new features.)

Reply:
I didn't add any types because I wanted to make it PHP 5.x possible. So what I'd like clarification on is: should I be adding any 7.0 features to this script? If so I'll go ahead and add types. Please clarify. Thanks

Reply:
> Per the comment at the top of the file, this script is currently compatible with PHP 7.0. We could discuss raising the constraint, but unlikely to go to 7.4. (The test runner can be invoked with a different PHP version than it is testing. I believe @TysonAndre also expressed a desire to use newer test runners on PECL extensions that support old PHP versions, so it's possible to use new features.)

The most significant new feature to me was the parallelism to take advantage of all cores on a test runner `-j N`.

I'm fine with raising the constraints - PECL projects which did need a run-tests.php that worked with 7.0-7.4 (and probably 8.x) could just copy the run-tests.php from the PHP-7.4 branch.

Reply:
This file is already incompatible with php 5.6 - I don't think anyone is planning to support php 5.6 in the latest versions of run-tests.php.

It aimed for 7.0+ in one release, but that's already available, and can get raised.

I don't have a personal need for newer features right now, but I could imagine others wanting it

```
    $environment = $_ENV ?? array();
```

Reply:
In that case i will update this code to give us the type safety advantages
of php 7. Thanks